QuEST/include/QuEST.h

```diff
@@ -1662,6 +1662,43 @@ int measureWithStats(Qureg qureg, int measureQubit, qreal *outcomeProb);
  */
 Complex calcInnerProduct(Qureg bra, Qureg ket);
 
+/** If both input states \p a and \p b are equal-size density matrices
+ * then computes the Hilbert-Schmidt scalar product between \p a and \p b, 
+ * which is equialent to the Frobenius inner product of matrices.
+ * That is, we define the Hilbert-Schmidt scalar product
+ * \f[
+    (( a, b ))_F = ((a, b))_HS := \text{Tr}[ a^\dagger b ] 
+ * \f]
+ * This is equivalent to the sum of products of matrix elemets, i.e.
+ * \f[
+    ((a, b))_HS = \sum\limits_i \sum\limits_j  conj(a_{ij}) b_{ij}
+ * \f]
+ * Note that if both \p a and \p b are density matrices of pure states
+ * \p bra and \p ket, then the equality holds
+ * \f[
+    ((a, b))_HS = |\langle \text{bra} | \text{ket} \rangle|^2
+ * \f]
+ * If both inputs are  equal-size state vectors then computes the inner product
+ * square \f$ |\langle \text{bra} | \text{ket} \rangle |^2\f$ via
+ * the usual inner product
+ * \f[
+    \langle \text{bra} | \text{ket} \rangle = \sum\limits_i  conj(bra_{i}) ket_{i}
+ * \f]
+ *
+ * @ingroup calc
+ * @param[in] bra qureg to be the 'bra' or 'a' (i.e. have its values conjugate transposed)
+ * in the inner product 
+ * @param[in] ket qureg to be the 'ket' or 'b' in the inner product 
+ * @returns the real Hilbert-Schmidt scalar product of density matrices
+            \p a and \p b or the absolute square of the inner product
+            between state vectors \p bra and \p ket 
+ * @throws exitWithError
+ *      if \p bra and \p ket or \p a and \p b have mismatching dimensions
+ *      or mismatching qureg types.
+ * @author Balint Koczor
```

Comment:
should be
```C
@author Balint Koczor (CPU)
```
unless you write the GPU code!

Reply:
Added (CPU) in acf7f95214544204939c8267381218a72ec98096 for now.